app/controllers/topics_controller.rb

```diff
@@ -538,6 +538,12 @@ def invite
       group_names: params[:group_names]
     )
 
+    if !guardian.is_staff? && topic.reached_recipients_limit?
```

Comment:
I feel like the responsibility of checking if there are too many recipients should be part of `Topic#invite` instead.

Reply:
I did initially put the code in the model method, but then I wanted to show a message to users when they try to go above the recipients limit and here is where I got a problem. We have `guardian.ensure_can_invite_to!(..)` called in the controller `invite` method, and this guardian method does check if there are too many recipients, so when it fails it's going to raise a generic 403 error. Is it OK if I repeat the code in the model method? I can't think of a cleaner way to achieve that.